### PR TITLE
bitswap nits

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

**Comment by magik6k** [comment]
Test fail on Jenkins looks unrelated:

```
ok 11 - 'ipfs daemon' succeeds

expecting success: 
    test_wait_for_file 20 100ms "$IPFS_PATH/api"
  
Error: timed out waiting for file: /go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0045-ls.sh/.ipfs/api
not ok 12 - api file shows up
#	
#	    test_wait_for_file 20 100ms "$IPFS_PATH/api"
#	
```

File: exchange/bitswap/bitswap.go [hunk 1]
```diff
@@ -372,16 +372,6 @@ func (bs *Bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg
 		return
 	}
 
-	// quickly send out cancels, reduces chances of duplicate block receives
-	var keys []*cid.Cid
```
Comment (Stebalien) [inline exchange/bitswap/bitswap.go]: @whyrusleeping this code was dead. We're currently sending out cancels from within sessions. However, this *does* mean that the above comment is no longer correct (it'll likely take a bit longer to send out cancels).

**Comment by Stebalien** [comment]
This should hopefully make gc a bit happier when bitswapping.

File: exchange/bitswap/decision/engine.go [hunk 2]
```diff
@@ -336,12 +337,12 @@ func (e *Engine) numBytesReceivedFrom(p peer.ID) uint64 {
 // ledger lazily instantiates a ledger
 func (e *Engine) findOrCreate(p peer.ID) *ledger {
 	e.lock.Lock()
+	defer e.lock.Unlock()
```
Comment (whyrusleeping) [inline exchange/bitswap/decision/engine.go]: I do want to note that this may cause a perf degradation

**Pull request merged**